### PR TITLE
Compute GLONASS sat states

### PR DESCRIPTION
**Pull request by jtec**
Fixes part of https://github.com/jtec/prx/issues/38 by adding GLONASS support.

To accommodate GLONASS

- Instead of transforming RINEX navigation file timestamps (e.g. clock reference time) into pandas Timedeltas by subtracting a constellation-specific epoch, we keep the timestamps as pandas Timedtamps, as GLONASS time has no epoch

### CODE REVIEW

**Comment by jtec** [comment]
@plutonheaven I still see around 4 km of GLONASS position offset w.r.t. SP3: 

```
FAILED                                                                   [100%]
     sv    clock_m  dclock_mps    dx_mps    dy_mps    dz_mps query_time_isagpst          x_m          y_m          z_m  sp3_velocity  diff_xyz_l2_m  diff_dxyz_l2_mps
0   C03  25.353899    0.000016 -0.000081 -0.000015  0.000034             0 days   -10.768271    -4.925024     0.886236     83.236451      11.874213          0.000089
1   C05  22.467530   -0.000003 -0.000077 -0.000014 -0.000024             0 days     2.836287    -3.659981     0.760886     29.139027       4.692434          0.000081
2   C30   9.119176    0.000012 -0.000046 -0.000006  0.000111             0 days    -0.977266    -0.586954    -0.313945   2882.927093       1.182424          0.000121
3   C38  21.175381    0.000009 -0.000088 -0.000006  0.000025             0 days     0.287980    -1.580404     1.052369   2902.155696       1.920440          0.000092
4   E24   0.757594    0.000012  0.000076  0.000008 -0.000002             0 days    -0.087363     0.461163    -0.541410   2505.110016       0.716539          0.000076
5   E30   0.827131   -0.000013 -0.000050 -0.000135  0.000092             0 days     0.399739    -0.715436     0.607427   2909.005230       1.020102          0.000171
6   G12 -11.322657   -0.000112  0.000042 -0.000055 -0.000094             0 days     0.752269    -0.537973     0.485160   3209.357264       1.044368          0.000116
7   G15 -11.649048   -0.000119 -0.000002 -0.000050 -0.000047             0 days     0.459807    -0.418800    -0.345475   2953.124956       0.711455          0.000069
8   G15 -11.649167   -0.000119 -0.000001 -0.000050 -0.000047             0 days     0.459806    -0.418850    -0.345522   2953.176109       0.711507          0.000069
9   J02   0.927012    0.000008  0.000080 -0.000109  0.000001             0 days     1.727331    -0.126516     0.052038   2217.035841       1.732739          0.000135
10  J03   0.907296   -0.000008  0.000030  0.000065 -0.000005             0 days    -0.137307    -0.541542     0.624522   1361.567745       0.837943          0.000071
11  R04  -4.060983   -0.000101  0.697876  0.021380 -0.383496             0 days  -332.767130  1819.784669 -4327.976151   3168.471931    4706.774667          0.796591
12  R05  -6.530216   -0.000019  0.255365  0.374125 -0.701884             0 days -2860.300572   839.883932 -3128.232060   3370.442376    4321.175789          0.835358


```

**Comment by jtec** [comment]
Confirmed that the error gets larger as integration times increases - at the ephemeris reference time the error is meter-level.

**Comment by jtec** [comment]
Fixed a bug in the k4 Runge-Kutta term, sp3 offset now down to 120 meters, after 24 minutes integration time, error exclusively in Z.

**Comment by jtec** [comment]
Computing `x_dot` the same way https://github.com/tomojitakasu/RTKLIB/blob/master/src/ephemeris.c#L261 does it, we end with reasonable offsets w.r.t. SP3.

So we have 3 different ways to compute the velocity derivatoive: The Springer Handbook, the ICD and RtkLib. Will need to dig deeper to figure out what almost everybody gets wrong here.

File: src/prx/main.py [hunk 1]
```diff
@@ -253,41 +253,43 @@ def format_flat_rows(row):
...
-        per_sat["time_of_reception_in_receiver_time"]
-        - per_sat.system_time_scale_epoch
-        - tof_dtrx
+    per_sat[
+        "time_of_emission_in_satellite_time_integer_second_aligned_to_receiver_time"
+    ] = per_sat["time_of_reception_in_receiver_time"] - tof_dtrx
+    # As error terms are tens of nanoseconds here, and the receiver clock is integer-second aligned to GPST, we
+    # already have times-of-emission that are integer-second aligned GPST here.
+    per_sat["time_of_emission_isagpst"] = (
+        per_sat.time_of_emission_in_satellite_time_integer_second_aligned_to_receiver_time
```
Comment (plutonheaven) [inline src/prx/main.py:275]: There is two similar columns: `time_of_emission_isagpst` and `time_of_emission_in_satellite_time_integer_second_aligned_to_receiver_time`, so we may drop one.

File: src/prx/main.py [hunk 2]
```diff
@@ -253,41 +253,43 @@ def format_flat_rows(row):
...
     )
-    per_sat["time_of_emission_weeksecond_system_time"] = per_sat.apply(
+    per_sat["time_of_emission_weeksecond_isagpst"] = per_sat.apply(
```
Comment (plutonheaven) [inline src/prx/main.py:277]: I would suggest to rename this parameter `time_of_emission_in_seconds_of_week_isagpst`.

**Review by plutonheaven — APPROVED** [review]

File: src/prx/main.py (hunk 1 above)
Comment (jtec) [reply src/prx/main.py:275]: Missed teh comment before merging , will do in https://github.com/jtec/prx/issues/73

File: src/prx/main.py (hunk 2 above)
Comment (jtec) [reply src/prx/main.py:277]: See https://github.com/jtec/prx/pull/70#discussion_r1554713805

**Pull request merged**